resources/views/docs/1/digging-deeper/child-process.md

```diff
@@ -0,0 +1,300 @@
+
+```php
+ChildProcess::message('Hello, world!', 'tail');
+```
+
+The message format and how they are handled will be determined by the program you're running.
+
+## Handling output
+A Child Process may send output via any of the following interfaces:
+
+- Its standard output stream (`STDOUT`).
+- Its standard error stream (`STDERR`).
+- A custom interface, e.g. a network socket.
+
+### Listening for Output (`STDOUT`)
+You may receive standard output for a process by registering an event listener for the
+[`MessageReceived`](#codemessagereceivedcode) event:
+
+```php
+use Illuminate\Support\Facades\Event;
+use Native\Laravel\Events\ChildProcess\MessageReceived;
+
+Event::listen(MessageReceived::class, function (MessageReceived $event) {
+    if ($event->alias === 'tail') {
+        info($event->data);
+    }
+});
+```
+
+### Listening for Errors (`STDERR`)
+You may receive standard errors for a process by registering an event listener for the
+[`ErrorReceived`](#codeerrorreceivedcode) event:
+
+```php
+use Illuminate\Support\Facades\Event;
+use Native\Laravel\Events\ChildProcess\ErrorReceived;
+
+Event::listen(ErrorReceived::class, function (ErrorReceived $event) {
+    if ($event->alias === 'tail') {
+        info($event->data);
+    }
+});
+```
+
+## Stopping a Child Process
+Your child processes will shut down when your application exits. However, you may also choose to stop them manually or
+provide this control to your user.
+
+If you have a `Native\Laravel\ChildProcess` instance, you may call the `stop` method on it:
+
+```php
+$tail->stop();
+```
+
+Alternatively, you may use the `ChildProcess` facade to stop a process via its alias:
+
+```php
+ChildProcess::stop('tail');
+```
+
+This will attempt to stop the process gracefully. The [`ProcessExited`](#codeprocessexitedcode) event will be
+dispatched if the process exits.
+
+## Restarting a Child Process
+As a convenience, you may simply restart a Child Process using the `restart` method. This may be useful in cases where
+the program has become unresponsive and you simply need to "reboot" it.
+
+If you have a `Native\Laravel\ChildProcess` instance, you may call the `restart` method on it:
+
+```php
+$tail->stop();
+```
+
+Alternatively, you may use the `ChildProcess` facade to restart a process via its alias:
+
+```php
+ChildProcess::restart('tail');
+```
+
+## Events
+
+NativePHP provides a simple way to listen for Child Process events.
+
+All events get dispatched as regular Laravel events, so you may use your `EventServiceProvider` to register listeners.
+
+```php
+protected $listen = [
+    'Native\Laravel\Events\ChildProcess\MessageReceived' => [
+        'App\Listeners\MainLoopEvent',
+    ],
+];
+```
+
+Sometimes you may want to listen and react to these events in real-time, which is why NativePHP also broadcasts all
+Child Process events to the `nativephp` broadcast channel.
+
+To learn more about NativePHP's broadcasting capabilities, please refer to the
```

Comment:
Maybe a simple example to illustrate frontend listening could be helpful here. This easy is skim over

Reply:
added in https://github.com/NativePHP/nativephp.com/pull/54/commits/08e0cd0ebc1ada409364df7a989f359ad095e9fd